solr/core/src/java/org/apache/solr/util/FileTypeMagicUtil.java

```diff
@@ -60,6 +60,14 @@ public static void assertConfigSetFolderLegal(Path confPath) throws IOException
         new SimpleFileVisitor<>() {
           @Override
           public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) {
+            if (Files.isSymbolicLink(file)) {
```

Comment:
Ah so much easier then when I had to detect symlinks for Apache Ant back in the day :) (https://bz.apache.org/bugzilla/show_bug.cgi?id=1550#c14)

However, I'm not sure it's valid to assume the user never wants to follow symbolic link. I can certainly imagine cases where folks would use a symlink for things they wanted to keep synced across config sets. It seems to me the correct behavior is to A) check for loops and fail if the symlinks are circular, and B) unroll the symlink (possibly emitting a warning) so that the contents are represented as the same tree that a person navigating with a shell would perceive.

Reply:
^ me on wrong browser again

Reply:
So, right now symbolic links don't work, they blow up with a cryptic error.  This PR replaces the cryptic error with a nicer error!   I think your suggestion is interesting, and if we wanted to support symbolic link lookup, that sounds like a nice addition to do in a seperate PR that removes the limitation.   Does that sound reasonable?